### ISSUE TITLE
rule proposal: forbid element/components

### ISSUE THREAD

**Issue by kentor** [yannickcr/eslint-plugin-react#887]
It would be useful to have a rule that forbids elements/components from a list of names.

In my organization we want to ban using `<button>` in favor of a custom `<Button>` component that is simply:

``` jsx
Button = (props) => <button type="button" {...props}>{props.children}</button>
```

The reason is to prevent a bug in IE where pressing enter on an input could trigger the click event on a button even when they don't share a parent form if the button has `type="submit"` (default if no `type`). (https://www.tjvantoll.com/2013/05/22/why-are-enter-keypresses-clicking-my-buttons-in-ie/)

**Comment by ljharb** [comment]
While a generic rule could be useful, I think a specific rule that requires that `<button>` have an explicit "type" defined would also be helpful.

**Comment by kentor** [comment]
I can see that being useful as well but it seems like those rules have different use cases and they can both exist at the same time. 

The reason I chose to use a custom Button component instead of enforcing a type prop was because
1.  I simply think type=submit is a bad default, and rather it be type=button. but specs are specs
2. we're enforcing 80 characters per line, so not having to specify the type is a big win

**Comment by ljharb** [comment]
Totally, I think adding both is good.

**Comment by kentor** [comment]
Great! I can start on forbidding elements

**Comment by alexzherdev** [comment]
I think `button-has-type` plus #890 address both of these.
